geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java

```diff
@@ -1711,14 +1711,13 @@ boolean removeTreeNode(TreeNode<K> p) {
               else
                 sp.right = p;
             }
-            if ((s.right = pr) != null)
-              pr.parent = s;
+
+            pr.parent = s;
           }
           p.left = null;
           if ((p.right = sr) != null)
             sr.parent = p;
-          if ((s.left = pl) != null)
-            pl.parent = s;
+          pl.parent = s;
```

Comment:
Why is it okay to delete the assignment (s.left = pl)?

Reply:
I made a mistake... 
Used the intelliJ auto-correct.
--fixed